QuasistaticBrownianThermalNoise/QuasistaticBrownianThermalNoise.cpp

```diff
@@ -1152,6 +1336,60 @@ void ElasticProblem<dim>::output_results(const unsigned int cycle) {
 
       // store current quadrature point's Young's modulus
       double theYijkl = 0.;
+      double sYijkl = 0;
+
+      // If custom loss angles are specified, then we calculate the energy
+      // for the coating using the appropriate youngs tensor.
+      if (config["Mirror"]["CustomLossAngles"].as<bool>()) {
+        for (unsigned int q_point = 0; q_point < n_q_points; ++q_point) {
+          for (int i = 0; i < dim; ++i) {
+            for (int j = 0; j < dim; ++j) {
+              for (int k = 0; k < dim; ++k) {
+                for (int l = 0; l < dim; ++l) {
+                  local_U_this_energy = 0.5 * fe_values.JxW(q_point) *
+                                        solution_gradients[q_point][j][i] *
+                                        solution_gradients[q_point][l][k];
+
+                  // only perform the energy calculation for points above 0
+                  // since we are only interested in the coating.
+
+                  if (fe_values.quadrature_point(q_point)(dim - 1) > 0.) {
+                    sYijkl =
```

Comment:
Why not put the if here, so only the getyijkl line is different? Then less copied code